### ISSUE TITLE
feature request: check if bash regular expression is meaningless

### ISSUE THREAD

**Issue by e-kwsm** [koalaman/shellcheck#3252]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2236?
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
[[ $1 =~ ^.*$ ]]
[[ $2 =~ .?$ ]]
[[ $3 =~ ^.{0,3} ]]

[[ $4 =~ .+ ]]
```

#### Here's what shellcheck currently says:

No issues are detected.

#### Here's what I wanted or expected to see:

```
[[ $1 =~ ^.*$ ]]
   ^-- SCXXXX (error): This condition is always true.

[[ $2 =~ .?$ ]]
   ^-- SCXXXX (error): This condition is always true.

[[ $3 =~ ^.{0,3} ]]
   ^-- SCXXXX (error): This condition is always true.

[[ $4 =~ .+ ]]
   ^-- SC2236 (style): Use [[ -n $4 ]] instead.
```
